sdk/src/Core/Amazon.Runtime/Telemetry/Attributes.cs

```diff
@@ -0,0 +1,84 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System.Collections.Generic;
+
+namespace Amazon.Runtime.Telemetry
+{
+    /// <summary>
+    /// Represents a collection of key-value pairs for attributes.
+    /// </summary>
+    public class Attributes
+    {
+        private readonly List<KeyValuePair<string, object>> _attributes;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Attributes"/> class.
+        /// </summary>
+        /// <param name="attributes">An initial collection of key-value pairs to populate the attributes.</param>
+        public Attributes(IEnumerable<KeyValuePair<string, object>> attributes)
+        {
+            _attributes = new List<KeyValuePair<string, object>>(attributes);
```

Comment:
This allows for duplicate keys:

```
var attributes = new Attributes(new List<KeyValuePair<string, object>>
{
    new KeyValuePair<string, object>("name", "John"),
    new KeyValuePair<string, object>("name", "Fred"),
    new KeyValuePair<string, object>("age", 30),
    new KeyValuePair<string, object>("isStudent", true)
});
```

Which in turn will cause a problem with Set/Get only using the first one found.

Reply:
After updating the internal attributes to be a dictionary, the duplicates will replace the previous version, similar to the OTel tags behavior